pkg/debuginfo/debuginfo_file.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debuginfo
+
+import (
+	"debug/elf"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/parca-dev/parca-agent/pkg/buildid"
+	"github.com/parca-dev/parca-agent/pkg/objectfile"
+)
+
+var dwarfSuffix = func(s *elf.Section) string {
+	switch {
+	case strings.HasPrefix(s.Name, ".debug_"):
+		return s.Name[7:]
+	case strings.HasPrefix(s.Name, ".zdebug_"):
+		return s.Name[8:]
+	case strings.HasPrefix(s.Name, "__debug_"): // macos
+		return s.Name[8:]
+	default:
+		return ""
+	}
+}
+
+type debugInfoFile struct {
+	*objectfile.ObjectFile
+
+	dbgOnce      sync.Once
+	dbgErr       error
+	hasDebugInfo bool
+
+	localDbgOnce       sync.Once
+	localDbgErr        error
+	localDebugInfoPath string
+}
+
+func (f *debugInfoFile) LocalHostDebugInfoPath() (string, error) {
```

Comment:
can we rename this to just `LocalDebugInfoPath`? `LocalHost` strongly suggested to me as the reader that what we're dealing here has something to do with the hostname